common/Models/ExtensionAdaptiveCard.cs

```diff
@@ -20,11 +23,18 @@ public class ExtensionAdaptiveCard : IExtensionAdaptiveCard
 
     public string TemplateJson { get; private set; }
 
+    private readonly IThemeSelectorService _themeSelectorService;
+
+    private AdaptiveCardParseResult? _parseResult;
+
     public ExtensionAdaptiveCard()
     {
         TemplateJson = new JsonObject().ToJsonString();
         DataJson = new JsonObject().ToJsonString();
         State = string.Empty;
+
+        _themeSelectorService = Application.Current.GetService<IThemeSelectorService>();

```

Comment:
If `ExtensionAdaptiveCard` is being made with DI, then `IThemeSelectorService` can be an argument to the constructor.

Reply:
Do we have a ViewModel for this model? Ideally the view-model would be registered and the theme service can be injected into-it. Depending on the use case, but model classes may not necessarily be needed to be registered, and a view-model for that model may be sufficient if we need to interact with the model and pass things to it based on UI/View interactions or changes.

Note: `Application.Current` is not available in unit testing, hence making this model challenging to test.

> In the context of a .NET MAUI app that uses MVVM, a dependency injection container will typically be 
used for registering and resolving views, registering and resolving view models, and for registering 
services and injecting them into view models

Reference: https://dotnet.microsoft.com/en-us/download/e-book/maui/pdf

Reply:
I didn't notice this was a model.  my bad.

Reply:
I looked at this again and I didn't need it anymore for the card to respond to theme changes. Removed.